spring-integration-hazelcast/src/test/java/org/springframework/integration/hazelcast/inbound/HazelcastClusterMonitorInboundChannelAdapterTests.java

```diff
@@ -0,0 +1,227 @@
+				hazelcastInstance);
+	}
+
+	@Test
+	public void testMigrationEvent() {
+		hazelcastInstance3.getLifecycleService().terminate();
+		final Message<?> msg = cmChannel3.receive(TIMEOUT);
+		verifyMigrationEvent(msg);
+	}
+
+	@Test
+	public void testLifecycleEvent() throws InterruptedException {
+		hazelcastInstance2.getLifecycleService().terminate();
+
+		Message<?> msg = cmChannel4.receive(TIMEOUT);
+		verifyLifecycleEvent(msg, LifecycleState.SHUTTING_DOWN);
+
+		msg = cmChannel4.receive(TIMEOUT);
+		verifyLifecycleEvent(msg, LifecycleState.SHUTDOWN);
+	}
+
+	@Test
+	public void testClientEvent() {
+		testClientEventByChannelAndGroupName(cmChannel5, TEST_GROUP_NAME1);
+	}
+
+	@Test
+	public void testMultipleMonitorTypes() {
+		testDistributedObjectEventByChannelAndHazelcastInstance(cmChannel6,
+				hazelcastInstance);
+
+		testMembershipEvent(hazelcastInstance, cmChannel6, "testKey2", "testValue2");
+	}
+
+	private void testMembershipEvent(
+			final HazelcastInstance instance, final PollableChannel channel,
+			final String key, final String value) {
+		Member member = instance.getCluster().getMembers().iterator().next();
+		member.setStringAttribute(key, value);
+
+		Message<?> msg = channel.receive(TIMEOUT);
+		verifyMembershipEvent(msg, MembershipEvent.MEMBER_ATTRIBUTE_CHANGED);
+	}
+
+	private void testClientEventByChannelAndGroupName(final PollableChannel channel,
+			final String groupName) {
+		final HazelcastInstance client = getHazelcastClientByGroupName(groupName);
+
+		Message<?> msg = channel.receive(TIMEOUT);
+		verifyClientEvent(msg);
+
+		client.getLifecycleService().terminate();
+
+		msg = channel.receive(TIMEOUT);
+		verifyClientEvent(msg);
+	}
+
+	private void testDistributedObjectEventByChannelAndHazelcastInstance(
+			final PollableChannel channel, final HazelcastInstance hazelcastInstance) {
+		final String distributedObjectName = "Test_Distributed_Map";
+		final IMap<Integer, String> distributedMap = hazelcastInstance
+				.getMap(distributedObjectName);
+
+		Message<?> msg = channel.receive(TIMEOUT);
+		verifyDistributedObjectEvent(msg, DistributedObjectEvent.EventType.CREATED,
+				distributedObjectName);
+
+		distributedMap.destroy();
+
+		msg = channel.receive(TIMEOUT);
+		verifyDistributedObjectEvent(msg, DistributedObjectEvent.EventType.DESTROYED,
+				distributedObjectName);
+	}
+
+	private HazelcastInstance getHazelcastClientByGroupName(final String groupName) {
+		final GroupConfig groupConfig = new GroupConfig();
+		groupConfig.setName(groupName);
+		final ClientConfig cfg = new ClientConfig();
+		cfg.setGroupConfig(groupConfig);
+
+		return HazelcastClient.newHazelcastClient(cfg);
+	}
+
+	private void verifyMembershipEvent(final Message<?> msg, final int membershipEvent) {
+		assertNotNull(msg);
+		assertNotNull(msg.getPayload());
+		assertTrue(msg.getPayload() instanceof MembershipEvent);
+		assertEquals(((MembershipEvent) msg.getPayload()).getEventType(), membershipEvent);
```

Comment:
must be flipped. I mean you should exchange `expected` and `actual`. See `assertEquals` source code :smile: